File: python/900/96A.py
players = input()

target = players[0]
count = 1
flag = False
for player in players[1:]:
    if player == target:
        count += 1
    else:
        target = player
        count = 1
    if count == 7:
        flag = True
        break

if flag:
    print("YES")
else:
    print("NO")
        

File: c++/900/96A.cpp
#include <iostream>
#include <string>

using namespace std;

int main(){
    string s;
    cin >> s;

    int count = 0;
    char target = s[0];
    bool dangerous = false;

    for (char c : s){
        if(c == target){
            count++;
        }
        else{
            target = c;
            count = 1;
        }
        if(count == 7){
            dangerous  = true;
            break;
        }
    }

    if(dangerous){
        cout << "YES";
    }
    else{
        cout << "NO";
    }
}